### PR TITLE
Dev angela data preprocessing

### PR DESCRIPTION
**Pull request by sansangela**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- [Directory Structure](#directory-structure)
+- [To-Do](#to-do)
+
+
+## Getting Started
+To run the lstm-model (Work In Progress), run the following command:
+
+```bash
+pip3 install -r third_party/requirements.txt
+python3 social_lstm/main.py
```
Comment (yuanxuns) [inline Readme.md:15]: 改成bazel command

File: social_lstm/main.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    # Load data
+    dataloader = DataLoader(
+        args.file_path,
+        args.dataset_name,
+        args.file_path_processed,
+        args.force_process_data_flag,
+    )
+    dataloader.reset_frame_ptr()
+
+    with tf.device(device):
```
Comment (yuanxuns) [inline social_lstm/main.py:94]: nit: tf2还需要specify device吗？

File: social_lstm/main.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        "--num_epochs", type=int, default=1, help="Number of training epochs"
+    )  # TODO: change later
+    parser.add_argument("--batch_size", type=int, default=5, help="Batch size")
+
+    args = parser.parse_args()
+
+    train(args)
+
+
+def train(args):
```
Comment (yuanxuns) [inline social_lstm/main.py:78]: 把train放到单独一个file

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,9 @@
+##
+# gitignore file for bazel projects
+##
+
+# Ignore all generated bazel symlinks.
+/bazel-*
+/bazel_tutorial
+*.bazel
+*.lock
```
Comment (yuanxuns) [inline .gitignore:9]: clang: add end of line

File: social_lstm/BUILD [hunk 5]
```diff
@@ -0,0 +1,4 @@
+py_binary(
+    name = "main",
+    srcs = ["main.py"],
+)
```
Comment (yuanxuns) [inline social_lstm/BUILD:4]: clang: add end of line

File: social_lstm/src/utils.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    """
+    num_seq, num_ped = len(y_true), len(ped_id_to_index_map)
+    y_true_dense_representation = np.zeros((num_seq, num_ped, 2))
+    for sequence_idx in range(len(y_true)):
+        indices = [ped_id_to_index_map[x] for x in y_true[sequence_idx][:, 0] if x in ped_id_to_index_map.keys()]
+        if not indices:
+            continue
+        y_true_dense_representation[sequence_idx, indices, :] = y_true[sequence_idx][:, 1:3]
+    
+    return -tf.math.reduce_mean(y_true_dense_representation-y_pred)
```
Comment (yuanxuns) [inline social_lstm/src/utils.py:24]: clang: add end of line

File: social_lstm/main.py [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+                        tf.zeros((num_peds_per_frame, args.hidden_dim))
+                    )
+                    lstm_cell = tf.Variable(
+                        tf.zeros((num_peds_per_frame, args.hidden_dim))
+                    )
+
+                    with tf.GradientTape() as tape:
+                        # Forward pass
+                        out = model(dense_representation_tf, lstm_hidden, lstm_cell)
+                        loss = mse_loss(target, out, ped_id_to_index_map) # TODO, placeholder
```
Comment (yuanxuns) [inline social_lstm/main.py:131]: nit: paper里的log likelyhood loss还没implement是吧？
